ui/app/components/raft-storage-overview.js

```diff
@@ -0,0 +1,67 @@
+import Component from '@ember/component';
+import { getOwner } from '@ember/application';
+import config from '../config/environment';
+import { inject as service } from '@ember/service';
+
+export default Component.extend({
+  flashMessages: service(),
+  useServiceWorker: null,
+
+  async init() {
+    this._super(...arguments);
+    if (this.useServiceWorker === false) {
+      return;
+    }
+    if ('serviceWorker' in navigator) {
```

Comment:
is this needed because not all browsers support `serviceWorkers`? could you help me understand the different between downloading via a service worker not? which method of downloading will be more common? i imagine the main benefit of download via a service worker is that it can happen in the background while the rest of the page is loading?

Reply:
Yep, you're correct about the check here, and this triggers the fallback mechanism. Which one will be more common depends on the setup of the UI (service worker APIs require https connections), and what browser you're using. This check is specifically for browser support. I'll add some comments about about why ServiceWorker is preferred here, but the main reason is that it offloads the download to the browser instead of the JS application. You could start the download and completely close the UI and it would still continue to download if you leave the browser open. The browser offload also gives you a progress bar where the in-memory one doesn't.